src/Common/src/System/Net/HttpKnownHeaderNames.TryGetHeaderName.cs

```diff
@@ -155,6 +155,7 @@ public static unsafe bool TryGetHeaderName(IntPtr buffer, int length, out string
                         case 'K': potentialHeader = KeepAlive; goto TryMatch;  // [K]eep-Alive
                         case 'S': potentialHeader = SetCookie; goto TryMatch;  // [S]et-Cookie
                         case 'U': potentialHeader = UserAgent; goto TryMatch;  // [U]ser-Agent
+                        case 'R': potentialHeader = RequestId; goto TryMatch;  // [R]equest-Id
```

Comment:
This headers are new, declared in this change. They are injected to propagate correlation context below in DiagnosticsHandler. They only make sense in Request and not passed in response.

I noticed 'TryGetHeaderName` function only called for response, but also all known headers including pure Request headers (e.g. User-Agent or Host) are handled there. 
I'm fine to remove new ones though.